vxportia/router.py

```diff
@@ -0,0 +1,92 @@
+
+from vumi.errors import DispatcherError
+from vumi.dispatchers.base import BaseDispatchRouter
+
+from twisted.internet import reactor
+from twisted.internet.protocol import Factory
+from twisted.internet.endpoints import clientFromString
+from twisted.internet.defer import inlineCallbacks
+
+from vxportia.protocol import PortiaProtocol
+
+
+class PortiaRouter(BaseDispatchRouter):
+    """
+    A router that connects to Portia, the number porting database,
+    to map incoming messages to MNOs and to determine how outbound
+    messages are meant to be routed back.
+    """
+
+    clock = reactor
+
+    @inlineCallbacks
+    def setup_routing(self):
+        self.portia_config = self.config['portia']
+
+        self.mno_transport_map = self.portia_config['mappings']
+        self.transport_mno_map = dict([
+            (transport_name, mno)
+            for mno, transport_name in self.mno_transport_map.items()])
+        self.default_transport = self.portia_config.get('default_transport')
```

Comment:
The idea was that for MSISDNs that we resolve to an MNO we've not configured a transport for. Now that I'm re-reading these we should probably nack them or raise an exception.